### PR TITLE
Update recipe to use wxwidgets cf package, and linux build to use gtk3

### LINKED ISSUE #49: Depend on wxwidgets (from conda-forge)
Now that [`wxwidgets`]( https://github.com/conda-forge/wxwidgets-feedstock ) is packaged in `conda-forge`. It would be good to add it as a dependency.

cc @conda-forge/wxwidgets (for vis)

### PR DESCRIPTION
**Pull request by pauldmccarthy**
This PR updates the wxPython recipe to build against the [conda-forge wxwidgets package](https://github.com/conda-forge/wxwidgets-feedstock/) instead of building its own copy of wxwidgets. As a by-product of this change, the Linux wxPython package will now use GTK3. 

Closes #49 and #82

> Original PR description
>
> This PR updates the Linux build to use GTK3, and adjusts dependencies so that the `wxGLCanvas` class is compiled to use EGL rather than GLX.
> 
> @ocefpaf @hmaarrfk (and anybody else interested in this recipe) do you have any thoughts on dropping GLX in favour of EGL? By my understanding, I think that the only real drawbacks will be that this build won't work on old systems that don't support EGL, and won't work for remote display via X11/SSH.  However, I may be misunderstanding the situation, so there may be some other drawbacks that I'm not seeing.
> 
> Whether to use EGL or GLX is presently a [compile-time choice](https://github.com/wxWidgets/Phoenix/issues/1923), but the wxWidgets developers have been [discussing making this a runtime option](https://github.com/wxWidgets/wxWidgets/issues/22325), so this recipe could be adjusted again in the future to support both.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I'm not so interested in this recipe. 

I think it should ideally be using wx package I built for c and cop instead of rebuilding it itself. But I've moved on to qt.

**Comment by hmaarrfk** [comment]
> xWidgets that is very close to the same age as the Phoenix code

This is something that is intrinsically possible at conda-forge due to our run exports

wxwidgets-feedstock declares that a package that was built with Version `3.2.4` will be compatible with `3.2.6` but not `3.3.0` and would need to be rebuilt.
https://github.com/conda-forge/wxwidgets-feedstock/blob/main/recipe/meta.yaml#L21

**Comment by pauldmccarthy** [comment]
I don't fully understand exactly how tightly coupled wxPython is to a specific version of wxWidgets; it does at least appear to be the case that wxPython releases are based on tagged wxWidgets releases (e.g. [wxPython 4.2.1 is based on wxWidgets 3.2.2.1](https://github.com/wxWidgets/Phoenix/tree/wxPython-4.2.1/ext), [wxPython 4.2.0 on wxWidgets 3.2.0](https://github.com/wxWidgets/Phoenix/tree/wxPython-4.2.0/ext)), so worst case we could pin each wxPython release to a specific wxWidgets release.

In any case, this discussion is somewhat tangential to the topic of this PR and, while I'm not against the proposal, I'd prefer to get a GTK3/EGL build out sooner rather than later.

One small issue with the current wxWidgets feedstock is that it looks like it is compiling for GLX rather than EGL, although this wouldn't be difficult to change. I did consider the idea of having separate GLX/EGL build variants, but as I alluded to earlier, I'm not entirely sure of the benefits of keeping GLX, or the downsides of dropping it.

**Comment by pauldmccarthy** [comment]
@RobinD42 @swt2c @Metallicow If you have any investment in the wxPython conda-forge package, or any thoughts on the discussion here they would be much appreciated! I'm mostly curious as to whether there are any potential downsides to dropping GLX other than those I mentioned in the PR description.

**Comment by swt2c** [comment]
Hey @pauldmccarthy from my perspective, the main issues with switching to EGL are incompatibilities with other OpenGL libraries that may not support EGL (or can't support both EGL and GLX at the same time).  I don't know if that applies in the context of conda-forge, though.

It did apply in the context of the Debian packages, for example, so Debian's wxWidgets and wxPython are still compiled with GLX support.

**Comment by Metallicow** [comment]
Umm... @pauldmccarthy No comment. Basically what Scott described above
probably. I hate it when I have to write a book in comment as to why I have
to sometimes support a legacy option vs a glorified option. Most linuxes
are flat in nature but for example if you wanted to use a old windows OS ,
folks would sometimes place another installer in the installer buried deep
alongside other files because way back the Internet didn't exist. Maybe
7zip was vanilla or a modded version for whatever reasons. But yes it is
sometimes a pain in the ass to have to make a decision, but keeping all
options may mean you just voluntold maintaining on the next follower.
Whatever your decision might be... A short concise comment in the code with
a link is always a good thing vs. 100 other folks complaining why this or
that didn't work at some point in the time-space continuem.

**Comment by pauldmccarthy** [comment]
@swt2c @Metallicow Thanks for the comments; perhaps sticking with GLX is the safest option for the time being. I don't have a very good idea of the environments this package is most commonly installed into, so perhaps it's best not to rock the boat too much. I'll update the recipe when I next have some free time.

**Comment by ocefpaf** [comment]
> @swt2c @Metallicow Thanks for the comments; perhaps sticking with GLX is the safest option for the time being. I don't have a very good idea of the environments this package is most commonly installed into, so perhaps it's best not to rock the boat too much. I'll update the recipe when I next have some free time.

@pauldmccarthy we can try to ship a version with GTK3/EGL in a development branch so folks can test it out. My guess is that we are OKish in conda-forge.

**Comment by pauldmccarthy** [comment]
It'll be a few days before I can look into the Windows failures..

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,11 +1,19 @@
 @echo on
 setlocal EnableDelayedExpansion
 
+:: The wxpython build system searches for wxwidgets library files
```
Comment (hmaarrfk) [inline recipe/bld.bat:4]: Thanks for the detailed comment for future maintainers

**Comment by pauldmccarthy** [comment]
@hmaarrfk @ocefpaf assuming that the current pipeline passes, I think this is ready to merge. I've done rudimentary tests on macOS, Linux (Wayland and X11), Windows, and WSL with the built-in Wayland server.

As the Linux wxWidgets package is built against GTK3/X11/GLX, in order to use the `wx.glcanvas.GLCanvas` under Linux/Wayland and WSL/Wayland, the `GDK_BACKEND=x11` environment variable must be set. Apart from that it all looks good!

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

I just tried to remove a few more CDTs. They have been used inconsistently, but I did make an attempt in the past to cleanup their usage in the Qt recipes so its somewhat "fresh" in my mind.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,96 +19,52 @@ source:
...
+    - gtk3             # [linux]
     - numpy
-    - pango  # [linux]
-    - pathlib2
-    - pcre2  # [not win]
     - pillow
-    - pthread-stubs
-    - pypubsub  # [py3k]
-    - python.app  # [osx]
-    - requests
```
Comment (hmaarrfk) [inline recipe/meta.yaml:107]: some of these are python libraries, why is it ok to remove them now?

Comment (pauldmccarthy) [reply recipe/meta.yaml:107]: Just bringing the recipe up to date with respect to the upstream requirements: https://github.com/wxWidgets/Phoenix/blob/master/requirements/install.txt
 - wxPython vendors pypubsub: https://github.com/wxWidgets/Phoenix/blob/master/wx/lib/pubsub/README_WxPython.txt
 - `requests` is only used at build time (it's only referenced from `build.py`)
 - `pathlib2` is only used at build time, and only as a fall back if the stdlb `pathlib` is not present (i.e. for old versions of Python)

Comment (hmaarrfk) [reply recipe/meta.yaml:107]: Great. Thank you for confirming

**Pull request merged**